components/merkledb/src/proof_map_index/proof.rs

```diff
@@ -440,6 +440,28 @@ impl<K, V> MapProof<K, V> {
         debug_assert!(self.proof.windows(2).all(|w| w[0].path < w[1].path));
         self
     }
+
+    /// Creates MapProof from provided `proof` and `entries` vectors. Used to construct proof
+    /// after deserialization.
+    /// TODO: maybe it's better to use builder methods(but I don't think they should be exposed to public).
```

Comment:
I think current approach is better. Builder methods are associated with some `ProofMapIndex`, and deserialized proofs are not.

Just name method `from_raw_parts` (by analogy from `Vec`) to indicate that user must be really sure that he wants to use it.